dynamics/client.py

```diff
@@ -517,6 +537,7 @@ def delete(self, *, query: Optional[str] = None) -> None:
             url=query,
             headers={**self.default_headers("delete"), **self.headers},
         )
+        response.raise_for_status()
```

Comment:
See [this comment](https://github.com/MrThearMan/dynamics-client/pull/4/files#r1148326348).